FactSet/factset_fund_data.R

```diff
@@ -0,0 +1,198 @@
+  filter(report_date == data_timestamp) %>%
+  select(factset_fund_id, fsym_id, reported_mv) %>%
+  collect()
+
+# connect all the factset funds (where available) to details containing:
+# - fsym_id associated with each holding in fund
+# - reported market value associated with each holding
+data <- data %>%
+  select(fund_isin, factset_fund_id) %>%
+  left_join(own_fund_detail)
+
+# Sanity Check: all funds for which a factset_fund_id exists has complete
+# coverage on reported_mv (complete in the sense that every security has an
+# associated market value, doesn't necessarily mean each fund has comprehensive
+# coverage of all securities)
+data %>%
+  mutate(factset_fund_id_exists = !is.na(factset_fund_id)) %>%
+  filter(factset_fund_id_exists) %>%
+  mutate(mv_exists = !is.na(reported_mv)) %>%
+  group_by(factset_fund_id_exists) %>%
+  summarize(isin_count = n())
+
+# Using the following two issues and asset maps, we can get info on what
+# asset_class is associated with each holding
+ref_v2_issue_type_map <- tbl(factset_db, "ref_v2_issue_type_map")
+ref_v2_asset_class_map <- tbl(factset_db, "ref_v2_asset_class_map")
+
+issue_type_code_asset_class_desc <- ref_v2_issue_type_map %>%
+  left_join(ref_v2_asset_class_map, by = "asset_class_code") %>%
+  select(issue_type_code, asset_class_desc)
+
+fsym_id_asset_type <- tbl(factset_db, "own_v5_own_sec_coverage") %>%
+  left_join(
+    issue_type_code_asset_class_desc,
+    by = c("issue_type" = "issue_type_code")
+    ) %>%
+  select(fsym_id, asset_type = asset_class_desc) %>%
+  collect()
+
+# connect asset type to fund data
+data <- data %>%
+  left_join(fsym_id_asset_type) %>%
+  select(fund_isin, reported_mv, asset_type, fsym_id)
+
+# connect ISINs to the fsym_ids containing each
+data <- data %>%
+  left_join(sym_isin) %>%
+  select(fund_isin, holding_isin = isin, reported_mv, asset_type)
+
+# It's important to note that each fund may have different types of holdings:
+# Equity, Fixed Income, Private Equity, Cash, Other
+# We don't expect Private Equity, Cash or Other to necessarily have corresponding ISINs
+# However, the Fixed Income class could be an issue.
+
+# Let's look at coverage by asset_type
+data %>%
+  filter(!is.na(asset_type)) %>%
+  mutate(holding_isin_exists = !is.na(holding_isin)) %>%
+  group_by(fund_isin, asset_type, holding_isin_exists) %>%
+  summarise(reported_mv = sum(reported_mv)) %>%
+  group_by(fund_isin) %>%
+  mutate(total_mv = sum(reported_mv)) %>%
+  group_by(fund_isin, holding_isin_exists) %>%
+  mutate(percent_coverage = reported_mv / total_mv)
+
+# we can also look at how many funds have above 50% coverage by AuM
+coverage <- data %>%
+  filter(!is.na(asset_type)) %>%
+  mutate(has_isin = !is.na(holding_isin)) %>%
+  group_by(fund_isin, has_isin) %>%
+  summarise(reported_mv = sum(reported_mv)) %>%
+  group_by(fund_isin) %>%
+  mutate(coverage = reported_mv / sum(reported_mv)) %>%
+  filter(has_isin) %>%
+  select(fund_isin, coverage)
+
+# 617 funds by ISIN
```

Comment:
I think this line of code and the code block above is related to this comment from ur PR:

> Of the 949 funds that have factset_fund_ids, 620 ISINs have SOME coverage by holding ISIN

but what does this mean? that not all factset funds have holding information?